File: app/src/main/java/com/avidprogrammers/insurancepremiumcalculator/longterm_pt_car_upto1500.java
package com.avidprogrammers.insurancepremiumcalculator;

import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.net.ConnectivityManager;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Toast;

import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.AdView;

/**
 * Created by Abhishek on 26-Mar-17.
 */

public class longterm_pt_car_upto1500 extends AppCompatActivity implements ConnectivityReceiver.ConnectivityReceiverListener {

    ConnectivityReceiver conn;

    CheckingStatus checkingStatus;

    private static final String TAG = "pt_car_above1500";
    private AdView mAdView;

    @Override
    protected void onDestroy() {
        super.onDestroy();
        unregisterReceiver(conn);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        checkingStatus=new CheckingStatus();
        conn=new ConnectivityReceiver();
        IntentFilter intentFilter=new IntentFilter();
        intentFilter.addAction(ConnectivityManager.CONNECTIVITY_ACTION);
        registerReceiver(conn, intentFilter);
        checkfunction(longterm_pt_car_upto1500.this);

        setContentView(R.layout.longterm_pt_car_upto1500);
        Toast.makeText(this, "longterm_pt_car_upto1500", Toast.LENGTH_SHORT).show();
        getSupportActionBar().setTitle("Car Policy Type");
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        mAdView = (AdView) findViewById(R.id.adView);
        AdRequest adRequest = new AdRequest.Builder().build();
        mAdView.loadAd(adRequest);


        findViewById(R.id.lt_car_upto1500_bp).setOnClickListener(listener_lt_car_upto1500_bp);
        findViewById(R.id.lt_car_upto1500_pp).setOnClickListener(listener_lt_car_upto1500_pp);
    };


    @Override
    public boolean onSupportNavigateUp(){
        finish();
        return true;
    }

    View.OnClickListener listener_lt_car_upto1500_bp = new View.OnClickListener() {
        @Override
        public void onClick(View view) {
            Intent intent = new Intent(longterm_pt_car_upto1500.this, longterm_bp_car_upto1500.class);
            startActivity(intent);
        }
    };

    View.OnClickListener listener_lt_car_upto1500_pp = new View.OnClickListener() {
        @Override
        public void onClick(View view) {
            Intent intent = new Intent(longterm_pt_car_upto1500.this, longterm_pp_car_upto1500.class);
            startActivity(intent);
        }
    };

    //checking connectivity
    public void checkfunction(Context context){
        boolean isConnected=ConnectivityReceiver.isConnected();
        //notification(isConnected,lp_taxi_upto18pass.this);
        checkingStatus.notification(isConnected,context);

    }

    @Override
    protected void onResume() {
        super.onResume();
        MyApplication.getInstance().setConnectivityListener(this);
    }

    @Override
    public void onNetworkConnectionChanged(boolean isConnected) {
        checkfunction(this);
    }

}

File: app/src/main/java/com/avidprogrammers/insurancepremiumcalculator/longterm_pt_car_upto1500.kt
package com.avidprogrammers.insurancepremiumcalculator

import android.content.Context

import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.avidprogrammers.insurancepremiumcalculator.ConnectivityReceiver.ConnectivityReceiverListener
import com.google.android.gms.ads.AdView
import android.os.Bundle
import android.content.IntentFilter
import android.net.ConnectivityManager
import android.content.Intent

/**
 * Created by Abhishek on 26-Mar-17.
 */
class longterm_pt_car_upto1500 : AppCompatActivity(), ConnectivityReceiverListener {
    var conn: ConnectivityReceiver? = null
    var checkingStatus: CheckingStatus? = null
    private val mAdView: AdView? = null
    protected override fun onDestroy() {
        super.onDestroy()
        unregisterReceiver(conn)
    }

    protected override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        checkingStatus = CheckingStatus()
        conn = ConnectivityReceiver()
        val intentFilter = IntentFilter()
        intentFilter.addAction(ConnectivityManager.CONNECTIVITY_ACTION)
        registerReceiver(conn, intentFilter)
        checkfunction(this@longterm_pt_car_upto1500)
        setContentView(R.layout.longterm_pt_car_upto1500)
        //        Toast.makeText(this, "longterm_pt_car_upto1500", Toast.LENGTH_SHORT).show();
        getSupportActionBar()!!.setTitle("Car Policy Type")
        getSupportActionBar()!!.setDisplayHomeAsUpEnabled(true)

/*        mAdView = (AdView) findViewById(R.id.adView);
        AdRequest adRequest = new AdRequest.Builder().build();
        mAdView.loadAd(adRequest);*/findViewById<View>(R.id.lt_car_upto1500_bp).setOnClickListener(
            listener_lt_car_upto1500_bp
        )
        findViewById<View>(R.id.lt_car_upto1500_pp).setOnClickListener(listener_lt_car_upto1500_pp)
    }

    override fun onSupportNavigateUp(): Boolean {
        finish()
        return true
    }

    var listener_lt_car_upto1500_bp = View.OnClickListener {
        val intent = Intent(this@longterm_pt_car_upto1500, longterm_bp_car_upto1500::class.java)
        startActivity(intent)
    }
    var listener_lt_car_upto1500_pp = View.OnClickListener {
        val intent = Intent(this@longterm_pt_car_upto1500, longterm_pp_car_upto1500::class.java)
        startActivity(intent)
    }

    //checking connectivity
    fun checkfunction(context: Context?) {
        val isConnected: Boolean = ConnectivityReceiver.Companion.isConnected
        //notification(isConnected,lp_taxi_upto18pass.this);
        checkingStatus!!.notification(isConnected, context!!)
    }

    protected override fun onResume() {
        super.onResume()
        MyApplication.Companion.instance!!.setConnectivityListener(this)
    }

    override fun onNetworkConnectionChanged(isConnected: Boolean) {
        checkfunction(this)
    }

    companion object {
        private const val TAG = "pt_car_above1500"
    }
}